internal/core/command/put_test.go

```diff
@@ -73,3 +80,92 @@ func TestNewPutCommandBody(t *testing.T) {
 		t.Error("Failed to verify the request body contents")
 	}
 }
+
+func TestNewPutCommandContentType(t *testing.T) {
+	tests := []struct {
+		name            string
+		expectedHeaders map[string]string
+		req             *http.Request
+	}{
+		{
+			"cbor content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR}, http.MethodPut),
+		},
+		{
+			"json content type header propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON}, http.MethodPut),
+		},
+		{
+			"no content type header provided",
+			map[string]string{
+				clients.ContentType: ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: ""}, http.MethodPut),
+		},
+		{
+			"user-agent header not propagated",
+			map[string]string{
+				"User-Agent": ""},
+			newRequestWithHeaders(map[string]string{
+				"User-Agent": "User-Agent",
+			}, http.MethodPut),
+		},
+		{
+			"content-md5 header not propagated",
+			map[string]string{
+				"Content-MD5": ""},
+			newRequestWithHeaders(map[string]string{
+				"Content-MD5": "Content-MD5",
+			}, ""),
+		},
+		{
+			"json content type propagated, content-md5 header not propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       ""},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeJSON,
+				"Content-MD5":       "Content-MD5"}, http.MethodPut),
+		},
+		{
+			"cbor content type propagated, user-agent header not propagated, content-md5 header not propagated",
+			map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "",
+				"Content-MD5":       "",
+			},
+			newRequestWithHeaders(map[string]string{
+				clients.ContentType: clients.ContentTypeCBOR,
+				"User-Agent":        "User-Agent",
+				"Content-MD5":       "Content-MD5",
+			}, http.MethodPut),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var loggerMock = logger.NewMockClient()
+			ctx := context.Background()
+			putCommand, _ := NewPutCommand(
+				testDevice,
+				testCommand,
+				"Test Body",
+				ctx,
+				nil,
+				loggerMock,
+				tt.req)
+			actualHeaders := map[string]string{}
+			for k := range tt.expectedHeaders {
+				actualHeaders[k] = putCommand.(serviceCommand).Request.Header.Get(k)
+			}
```

Comment:
See my other comment regarding testing all the values present in `putCommand` rather than just the header present from the list of `expectedHeaders`